velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -73,28 +76,41 @@ class ParquetTpchTestBase : public testing::Test {
     auto task = assertQuery(tpchPlan, duckDbSql, sortingKeys);
   }
 
-  void TearDown() {
+  static void TearDownTestSuite() {
     connector::unregisterConnector(kHiveConnectorId);
-    unregisterParquetReaderFactory();
+    velox::parquet::unregisterParquetReaderFactory();
   }
 
  private:
   /// Write TPC-H tables as a Parquet file to temp directory in hive-style
   /// partition
-  void saveTpchTablesAsParquet() {
-    constexpr int kRowGroupSize = 10'000;
-    const auto tableNames = tpchBuilder_.getTableNames();
-    for (const auto& tableName : tableNames) {
+  static void saveTpchTablesAsParquet(memory::MemoryPool* pool) {
+    static constexpr auto tables = {
+        tpch::Table::TBL_PART,
+        tpch::Table::TBL_SUPPLIER,
+        tpch::Table::TBL_PARTSUPP,
+        tpch::Table::TBL_CUSTOMER,
+        tpch::Table::TBL_ORDERS,
+        tpch::Table::TBL_LINEITEM,
+        tpch::Table::TBL_NATION,
+        tpch::Table::TBL_REGION};
+
+    for (const auto& table : tables) {
+      auto tableName = std::string(toTableName(table));
       auto tableDirectory =
           fmt::format("{}/{}", tempDirectory_->path, tableName);
-      fs::create_directory(tableDirectory);
-      auto filePath = fmt::format("{}/file.parquet", tableDirectory);
-      auto query = fmt::format(
-          fmt::runtime(kDuckDbParquetWriteSQL_.at(tableName)),
-          tableName,
-          filePath,
-          kRowGroupSize);
-      duckDb_->execute(query);
+
+      auto tableSchema = velox::tpch::getTableSchema(table);
+      auto query = fmt::format("SELECT * FROM {}", tableName);
```

Comment:
not sure I follow. Why do we still need duckDB here? Couldn't we just generate the TPC-H data using TpchConnector and write them to a parquet file?

Reply:
If we write them to parquet, we need to enable the parquet extension in DuckDB to read which we want to avoid.
The tpch connector results in Velox are also not matching the tpch results in Duck since the date generated is different.
This is the simpler workaround. We use the same date from duck tpch extension, write to parquet, and compare velox tpch plans on parquet against duck tpch extension.

Reply:
@duanmeng Pedro and I discussed offline and we felt it is better to get off the Duck tpch extension.
I added an API to `tpch::getQuery()` here https://github.com/facebookincubator/velox/pull/6784
Can we go back to the Velox tpch connector approach to generate data? Then we can save the tables in Duck via `DuckDbQueryRunner::createTable`.

Reply:
OK, got it. I will update this PR once https://github.com/facebookincubator/velox/pull/6784 lands.

Reply:
Thank you guys

Reply:
@duanmeng the PR landed. Thanks!

Reply:
@majetideepak @pedroerp  I've updated this PR to use tpch connector approach.